lib/internal/url.js

```diff
@@ -475,9 +476,8 @@ class URLSearchParams {
     name = StringPrototypeToWellFormed(`${name}`);
     value = StringPrototypeToWellFormed(`${value}`);
     ArrayPrototypePush(this.#searchParams, name, value);
-    if (this.#context) {
-      this.#context.search = this.toString();
-    }
+
+    if (this.#context) setURLSearchParamsModified(this.#context);
```

Comment:
Can you please keep the brackets to minimize the diff? Smaller diffs come in handy when using `git blame`
```suggestion
    if (this.#context) {
      setURLSearchParamsModified(this.#context);
    }
```